mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/mosaic.kt

```diff
@@ -100,16 +98,14 @@ internal class MosaicComposition(
 	}
 }
 
-internal abstract class BaseMosaicComposition(coroutineScope: CoroutineScope) {
-	protected abstract val onRender: (CharSequence) -> Unit
-	protected abstract val rendering: Rendering
+internal abstract class MosaicComposition(coroutineScope: CoroutineScope) {
+	protected abstract val onEndChanges: (MosaicNode) -> Unit
 
 	private val job = Job(coroutineScope.coroutineContext[Job])
 	private val clock = BroadcastFrameClock()
 	protected val composeContext: CoroutineContext = coroutineScope.coroutineContext + job + clock
 
-	private val rootNode = createRootNode()
-	private val applier = MosaicNodeApplier(rootNode) { onRender(rendering.render(rootNode)) }
+	private val applier = MosaicNodeApplier(createRootNode()) { rootNode -> onEndChanges(rootNode) }
```

Comment:
```suggestion
	private val applier = MosaicNodeApplier(createRootNode(), onEndChanges)
```

Reply:
I also wanted to do this, but then Android Studio writes such a warning
<img width="800" alt="image" src="https://github.com/user-attachments/assets/3cc57ac9-4f7d-42e1-b744-9aaf599b1749">